src/Http/Client.php

```diff
@@ -118,7 +118,7 @@ public function request(
         $url = $endpoint . '?' . $full_query_string;
 
         // Wraps to passed client options.
-        $options = array_merge($this->clientOptions, $options);
+        $options = [...$this->clientOptions, ...$options];
```

Comment:
I was wondering if this was just a "use the latest thing" refactor or if there is a real benefit, and there is a benefit. The elimination of a function call, and better optimization at compile time. I am making a note of this. 

I would love to know how this compares to `[] + []` though. I couldn't find any documentation on that.

To really put this under a microscope, would it be more performant to spread our options into `$this->clientOptions` without creating a new array and overwriting the old one? Like so:

```php
array_push($this->clientOptions, ...$options);
```

But that would force us to use $this->clientOptions in the following lines, and would obfuscate the meaning of clientOptions as "preconfigured client options". 

Ok. I'm done :) 🤫

Reply:
For me, there is an element of "use the latest thing" but I would phrase it as motivation to "learn the new thing." I am thrilled to understand it more deeply though. Thank you for sharing!!

Reply:
I hope you don't mind me piggy-backing some learning for myself on top of learning from you. Thank you.

Reply:
I appreciate it! 😍